submodules/strategy/strategy.js

```diff
@@ -4528,6 +4026,212 @@ define(function(require) {
 				autoScroll: false,
 				title: self.i18n.active().strategy.addOfficeHours.title
 			});
+		},
+
+		strategyAddEditOfficeHolidaysPopup: function(args) {
+			var self = this,
+				callback = args.callback,
+				holidayRule = args.holidayRule ? args.holidayRule : {},
+				existingHolidays = args.existingHolidays,
+				isRecurring = _.get(holidayRule, 'holidayData.recurring', false),
+				getListOfYears = function getListOfYears() {
+					var date = new Date(),
+						year = parseInt(date.getFullYear()),
+						totalYears = 3,
+						yearsArray = [];
+
+					while (totalYears >= 0) {
+						yearsArray.push(year);
+						year++;
+						totalYears--;
+					}
+
+					return yearsArray;
+				},
+				getListOfDates = function getListOfDates() {
+					var date = 0,
+						datesArray = [];
+
+					while (date < 31) {
+						date++;
+						datesArray.push(date);
+					}
+
+					return datesArray;
+				},
+				formatMonth = function formatMonth(holidayRule, type) {
+					var month = _.get(holidayRule, 'holidayData.' + type);
+					return month ? month - 1 : 0;
+				},
+				dataToTemplate = _.merge({}, holidayRule, {
+					months: _.map(
+						self.months,
+						_.partial(monster.util.tryI18n, self.i18n.active().strategy.holidays.months)
+					),
+					ordinals: self.i18n.active().strategy.holidays.ordinals,
+					days: self.i18n.active().strategy.holidays.days,
+					years: getListOfYears(),
+					dates: getListOfDates(),
+					toMonth: formatMonth(holidayRule, 'toMonth'),
+					fromMonth: formatMonth(holidayRule, 'fromMonth')
+				}),
+				$template = $(self.getTemplate({
+					name: 'addEditOfficeHolidays',
+					data: dataToTemplate,
+					submodule: 'strategy'
+				})),
+				popup;
+
+			if (!_.isEmpty(holidayRule)) {
+				var selectedType = holidayRule.holidayType;
+
+				$template
+					.find('#date_type')
+					.val(holidayRule.holidayType);
+
+				$template
+					.find('.row-fluid')
+					.removeClass('selected');
+
+				$template
+					.find('.row-fluid.' + selectedType)
+					.addClass('selected');
+
+				if (selectedType === 'single' && isRecurring) {
+					$template
+						.find('.single .year')
+						.addClass('hide');
+				}
+			}
+
+			$template.find('.cancel').on('click', function(event) {
+				event.preventDefault();
+				popup.dialog('close');
+			});
+
+			$template.find('#name').on('keyup', function(event) {
+				var $this = $(this),
+					isEmpty = _.isEmpty($this.val());
+
+				$template.find('.no-name-error')[isEmpty ? 'slideDown' : 'slideUp'](200);
+				$template.find('.minimum-name-error').slideUp(200);
+				$template.find('.maximum-name-error').slideUp(200);
+				$template.find('.duplicate-name-error').slideUp(200);
```

Comment:
This could be shortened a bit:
```suggestion
				$template.find('.no-name-error')[isEmpty ? 'slideDown' : 'slideUp'](200)
					.siblings().slideUp(200);
```